File: analyzers/tests/SonarAnalyzer.Test/TestCases/Hotspots/RequestsWithExcessiveLength_CustomValues.cs
using System;
using Microsoft.AspNetCore.Mvc;

namespace Tests.Diagnostics
{
    public class MyController : Controller
    {
        [HttpPost]
        [RequestSizeLimit(43)] // Noncompliant {{Make sure the content length limit is safe here.}}
//       ^^^^^^^^^^^^^^^^^^^^
        public ActionResult PostRequestAboveLimit()
        {
            return null;
        }

        [HttpPost]
        [RequestFormLimits(MultipartBodyLengthLimit = 43)] // Noncompliant {{Make sure the content length limit is safe here.}}
//       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        public ActionResult MultipartFormRequestAboveLimit()
        {
            return null;
        }

        [HttpPost]
        [RequestSizeLimit(41)]
        public ActionResult PostRequestBelowLimit()
        {
            return null;
        }

        [HttpPost]
        [RequestFormLimits(MultipartBodyLengthLimit = 41)]
        public ActionResult MultipartFormRequestBelowLimit()
        {
            return null;
        }
    }
}

File: analyzers/tests/SonarAnalyzer.Test/TestCases/Hotspots/RequestsWithExcessiveLength_CustomValues.vb
Imports System
Imports Microsoft.AspNetCore.Mvc

Namespace Tests.TestCases

    Public Class MyController
        Inherits Controller

        <HttpPost>
        <RequestSizeLimit(43)>
        Public Function PostRequestAboveLimit() As ActionResult
        '^^^^^^^^^^^^^^^^^^^^ Noncompliant@-1 {{Make sure the content length limit is safe here.}}
            Return Nothing
        End Function

        <HttpPost>
        <RequestFormLimits(MultipartBodyLengthLimit:=43)>
        Public Function MultipartFormRequestAboveLimit() As ActionResult
        '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Noncompliant@-1 {{Make sure the content length limit is safe here.}}
            Return Nothing
        End Function

        <HttpPost>
        <RequestSizeLimit(41)>
        Public Function PostRequestBelowLimit() As ActionResult
            Return Nothing
        End Function

        <HttpPost>
        <RequestFormLimits(MultipartBodyLengthLimit:=41)>
        Public Function MultipartFormRequestBelowLimit() As ActionResult
            Return Nothing
        End Function

    End Class

End Namespace
